### ISSUE TITLE
Error value returned if JSON payload contains empty email field when dealing with dual memberships

### ISSUE THREAD

**Issue by deadlyraptor** [deadlyraptor/contact-copy#23]
As of commit 7e863c4, the agile.py script works except in once case. When a customer purchases a dual membership, they enter a second entry with the contact info for the second person. If the transaction is made via the web, an email address is required. But if the transaction is made at the box office, then while a second entry _is_ required, an email (and address, phone, etc.) is not required.

The spreadsheet for the membership sales activity contains all member entries, including every person part of a dual membership, e.g., if one transaction was for a single membership and a second transaction was for a dual, then the spreadsheet contains three rows.

The JSON payload generated using append_members() creates an entry with an empty value for the email_addresses dict key. When add_contacts() is called, it returns a 400 response (Bad Request) from Constant Contact with a message that email_addresses is of invalid type. In this case, the type is 'None' since the value is empty.

What is the best way to resolve this?

Options:
1. Skip the contact if the sequence number equals 2, which means the contact is part of a dual membership.
2. Add the contact using the same email address as the contact right before it, as that is the parent contact.

Option 1 is annoying but if the member chooses not to provide an email address for the second person, then it is what it is.

Option 2 may present a problem as the addcontacts API method for Constant Contact states the following: "In the JSON payload, contacts are identified by their email address in an import_data array (see the JSON Request example). If a contact in the array already exists in the user's account, the properties included in the request are updated, and any existing properties not included in the import are not changed (action is an update only, not a PUT, so no property values are erased if left blank)."

Basically, if I try to upload two contacts with the same email address, it'll be treated as the first one being updated with the info from the second instead of creating a brand new contact. It _is_ possible to add a contact into Constant Contact manually via the website with the same email address as another but it actually notices this and prompts the user to either edit the existing entry, assuming the user is actually updating it, or continue editing the one being uploaded. When submitted, there will be two entries in Constant contact with the same email address.

All this to say that we add members to Constant Contact in order to send them specific emails via the platform. Presumably, if the members are a couple, an email to one will suffice as he or she will inform the other party.

But still: what to do?

**Comment by deadlyraptor** [comment]
Going with option 1. The code to skip entries with no email addresses is:

```
for count in range(record_count):
    if collection[count][7].text is not None:
        append_members(collection[count])
```

**Comment by deadlyraptor** [comment]
Resolved by daa1696.
